components/service/glean/docs/metrics/testing-metrics.md

```diff
@@ -0,0 +1,68 @@
+# Unit testing glean metrics
+
+In order to support unit testing inside of client applications using glean, a set of testing API 
+functions have been included.  The intent is to make glean easier to test 'out of the box' in any
+client application it may be used in.  These functions expose a way to inspect and validate recorded
+metric values within the client application but are restricted to test code only through visibility
+annotations (`@VisibleForTesting(otherwise = VisibleForTesting.NONE)`).
+
+## General test API method semantics
+
+To check if a value exists (i.e. it has been recorded), there is a `testHasValue(pingName: String)`
```

Comment:
Can we note here already that `pingName` is optional?
Or simply use `testHasValue()` and don't mention `pingName` until later in the document?

Most users should not need to use `pingName` and might get distracted by this.